chore(perf): update table pagination to be page based

moo-onthelawn opened the pull request:
<!-- Featured screenshots/recordings -->

<!-- Overall purpose of the PR -->

Update our table pagination to be page based (instead of an infinite list where we just click "View More") - [designs](https://www.figma.com/design/fMQodZKzn9B5aZTN5G0fLJ/dYdX-%E2%80%BA-Desktop?node-id=24757-11376&t=29mBPDSvQCSzbumm-0).


https://github.com/dydxprotocol/v4-web/assets/70078372/40f0bb23-56f7-4ec1-876d-a580794f5e27



| Light Mode | Classic Mode | Dark Mode |
| ---- | ---- | ---- | 
| <img width="1324" alt="Screenshot 2024-05-14 at 3 07 33 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/a68c34da-fae6-419e-af2d-e7f110f72a7b"> | <img width="1366" alt="Screenshot 2024-05-14 at 3 08 17 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/de39b898-1a45-404a-aae2-481e18c4d671"> | <img width="1327" alt="Screenshot 2024-05-14 at 3 07 41 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/a4faf16d-e692-4c41-aff9-51fe57a994b5"> |

| Table Name | Table | Mobile | 
| ---- | ---- | ---- |
| Fills (Portfolio) | <img width="1067" alt="Screenshot 2024-05-14 at 3 15 20 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/393b3a06-0b49-4033-ac08-159439f4621b"> | <img width="967" alt="Screenshot 2024-05-14 at 3 15 32 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/6c3bb3d7-6a64-47fc-a1d7-d86b39b69bc2"> |
| Fills (Trade) | <img width="904" alt="Screenshot 2024-05-14 at 3 20 07 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/5dfbb0a3-ef7b-464b-b5fe-69fd347669a6"> | <img width="812" alt="Screenshot 2024-05-14 at 3 21 00 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/0fb357a8-7c8a-43c3-a53d-3ad6a8d387b2"> |
| Markets Dropdown | <img width="734" alt="Screenshot 2024-05-14 at 3 22 03 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/aba11e7e-c295-4dae-bda0-43e414061678"> | <img width="523" alt="Screenshot 2024-05-14 at 3 22 14 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/d0ab16f9-a813-45e0-a81d-35a7f3e0d5b3"> |
| Markets Compact | <img width="435" alt="Screenshot 2024-05-14 at 3 13 53 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/af3b9321-a18c-4381-baba-b7d9d2e16523"> | <img width="865" alt="Screenshot 2024-05-14 at 3 13 41 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/0400ecc5-156e-4162-9c5b-5ec4e91824bc"> |
| Markets | <img width="1341" alt="Screenshot 2024-05-14 at 3 10 02 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/c9bc600b-d3cf-45b2-ac61-ae4bfb42ec79"> | <img width="876" alt="Screenshot 2024-05-14 at 3 10 43 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/d1e465b9-dcd0-430f-8057-633ac7ad0e69"> |
| TransferHistory | <img width="1284" alt="Screenshot 2024-05-14 at 3 16 26 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/d21e661a-8de7-4b57-b4c2-ff6e17716aec"> | <img width="967" alt="Screenshot 2024-05-14 at 3 16 32 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/31750fd6-6bc8-4714-aeae-f87e287f3e92"> |
| FeeTable | <img width="1346" alt="Screenshot 2024-05-14 at 3 19 09 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/9de72c4e-47ae-4b7c-8526-642deef4bd35"> | <img width="932" alt="Screenshot 2024-05-14 at 3 19 22 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/fbc8f2d9-8caf-4672-938c-8ccb2ec5a501"> |




---

<!-- Reorder/delete the following sections accordingly: -->

## Views

* `Portfolio`
  * Update all tables on this page to have an initial page size of `20` (large pages)
  
## Components
* Pass through `initialPageSize` props in `FundingPaymentsTable`, `TransferHistoryTable`
* New: `TablePaginationRow`
  * New component for table pagination; chose 5, 10, 15, 20 as possible page sizes
  * The pagination buttons look as follows:
  
| < 7 items | >7 items; is one of first or last 2 items | > 7 items |
| ---- | ---- | ---- |
| <img width="1349" alt="Screenshot 2024-05-14 at 3 06 02 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/f6d1e4f5-be50-4ec3-be89-5aced3036757"> | <img width="1333" alt="Screenshot 2024-05-14 at 3 06 09 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/4abc8b46-9b88-474a-a581-b447f2ffc2ce"> | <img width="1341" alt="Screenshot 2024-05-14 at 3 06 15 PM" src="https://github.com/dydxprotocol/v4-web/assets/70078372/4e2011ac-f49b-4642-bd3f-d5e146935a2d"> |
  
* `Button`
  * Updated disabled state of `navigation` button (it doesn't usually have a border or background, so shouldn't have one when disabled either)
* `Table`
  * Updated `viewMoreConfig` prop to be just `initialPageSize`
  * Removed a bunch of commented out code (I don't think we'll ever uncomment this)
  * Update from "view more" button to use newly created `TablePaginationRow`
* `DropdownSelectMenu`
  * Allow configuration of font size
* `OrderbookTradesTable` 
  * Make sure this table never paginates
  
## Styles/Mixins

* `Table`
  * Rename `tableHeader-textColor` to be `--tableStickyRow-textColor`
  * Rename `tableHeader-backgroundColor` to be `--tableStickyRow-backgroundColor`

## Hooks

* `hooks/useTablePagination`
  * new hook that handles maintaining/setting state of page size, current page, and pagination buttons to display


## Packages

* `localization`
  * updated: v1.1.82 -> v1.1.84
  
## Testing
* Tested on API trader simulation (verified that table updates, but also doesn't crash performance)
---

<!-- Additional screenshots/recordings, before/after comparisons, testing instructions etc. -->

moo-onthelawn commented on src/components/Button.tsx:
```diff
@@ -129,34 +129,46 @@ const buttonActionVariants = {
       // Ordered from lowest to highest priority (ie. Disabled should overwrite Active and Loading states)
-      ${state[ButtonState.Loading] && buttonStateVariants[ButtonState.Loading]}
-      ${state[ButtonState.Disabled] && buttonStateVariants[ButtonState.Disabled]}
+      ${state[ButtonState.Loading] && buttonStateVariants(action)[ButtonState.Loading]}
```
sort of a weird mishmash of using function arguments + indexing into record; I think I prefer function arguments (can update the record) - but open to other ideas if there's a better way of doing it

moo-onthelawn commented on src/views/MarketsDropdown.tsx:
```diff
@@ -361,6 +364,14 @@ Styled.Table = styled(Table)`
 
+  tfoot {
+    --stickyArea-totalInsetTop: 0px;
+    --stickyArea-totalInsetBottom: 3px; //xcxc
```
ok I know this has to be 3px for possible rendering, but unsure what variables I'm supposed to be setting to get this properly calculated 🫠 would love to pair on this one because everything else looks fine

aforaleka commented:
the pagination component looks great!!!

some initial feedback just testing it out:
<img width="1193" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/909e418a-7a6b-4e1b-90b7-70e4f3f7a5c3">
- seems like the last row on the page has some weird padding going so it's indented

<img width="1194" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/acee97bd-3d05-4961-9612-3fb554d32a6e">
when there's only one item it's stretched -- is it intended?

<img width="1180" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/f96f41a1-6ce2-480f-8a14-2d779327293e">
nit: can we only show the pagination if there are more than 5 (lowest possible) items? 

<img width="174" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/1c14424b-22df-4dc9-8e5d-a21d3dabc21b">
nit: should we match the font size of the dropdown to the trigger one?

<img width="1455" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/6c8abe11-79f5-41c5-9759-f25342292fff">
i think tables with set length / number of rows, we should just show everything at once, e.g. the fees table will benefit from no pagination

<img width="1306" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/418837e7-35ba-424a-b2a9-48f3041821f2">
not sure if existing bug, but if i search click 2nd page, search shiba, and then empty the search bar, i'm stuck at just shiba
<img width="1306" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/c4319d04-1821-4bdb-8d71-c1d9d8806570">

<img width="714" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/013aacfa-ee0c-4ff4-b464-b6bf367a3828">
existing bug -- can we remove pagination from order book table for mobile?

moo-onthelawn commented:
whee ty for such a thorough review!!! (and catching all these bugs I didn't notice)
 
> * seems like the last row on the page has some weird padding going so it's indented;

fixed!

> there's only one item it's stretched -- is it intended?

hmm I wasn't able to repro this... lemme try a lil harder

> nit: can we only show the pagination if there are more than 5 (lowest possible) items?

👍 

> nit: should we match the font size of the dropdown to the trigger one?

done!

> i think tables with set length / number of rows, we should just show everything at once, e.g. the fees table will benefit from no pagination

ooh good point, added a prop in Table to allow callers to disable pagination

> not sure if existing bug, but if i search click 2nd page, search shiba, and then empty the search bar, i'm stuck at just shiba

hmm also unable to repro, lemme try again

>  existing bug -- can we remove pagination from order book table for mobile?

oh smart - also noticed it was showing on trades (and it shouldn't be), removed

aforaleka commented on src/components/DropdownSelectMenu.tsx:
```diff
@@ -125,12 +125,15 @@ const $DropdownIcon = styled.span`
 `;
 
 const $Content = styled(Content)`
+  --dropdownSelectMenu-item-font-size: inherit;
```
yeeee
